1163/non intrusive price suggestion

alfetopito opened the pull request:
## Update July 2

- Added swap icon
- No longer showing `infinity` prices
- Added tooltip. Please validate text contents:
![screenshot_2020-07-02_13-53-28](https://user-images.githubusercontent.com/43217/86408316-a12fac00-bc6b-11ea-9775-51f7bde9359c.png)


## Update Jun 30

- Did a minimal styling, properly aligning things and matching other parts of the app
- Adjusted widget height to fit the new entry
- No longer displaying `infinity` for bad prices. Using `N/A` instead
- Disabling input when `N/A`
- Added base/quote token suffix to be explicit on what this price refers too (although being a bit redundant)
- Added `selling` word to sentence to make it clear what the amount refers to: `Onchain orderbook price for selling <amount> <symbol|name|address>`
- Fixed the displayed symbol in the sentence. Previously it was changing every time the price was inverted
- Displaying second price estimation only when amount is different from default of 1 unit. 

Worst case scenario (long token symbols):
![screenshot_2020-06-30_14-49-37](https://user-images.githubusercontent.com/43217/86181517-fd1af900-bae2-11ea-8b36-b32d2ae261d9.png)

Regular use case (short token symbols):
![screenshot_2020-06-30_15-05-39](https://user-images.githubusercontent.com/43217/86181570-20de3f00-bae3-11ea-9362-74a502000ada.png)

### TODO:
- [ ] Show % difference between prices (slippage?)
- [x] Add tooltips
- [x] Add swap price buttons

## Proposal

No longer automatically update price inputs with price suggestion.
Instead, show a button and allow the user to make the choice.

Closes #1163

This change adds mostly the logic of fetching the prices and displaying it in a button.

I added 1 price for a single unit and one price for the selected amount.
This way the user has a reference of the `slippage` given his amount.

When switching to the inverse price, the suggestion is also inverted.
When clicking in a button, both prices are filled.

![screenshot_2020-06-29_15-09-52](https://user-images.githubusercontent.com/43217/86060981-b3ff7200-ba1a-11ea-8d09-f8a566c69708.png)

It's hideous, I know. Will work on that.

As always, open for suggestions/feedback.

## TODO:
- [x] Style
- [x] Do not show `infinity` when one of the prices is too big/too small
- [x] More testing, look for edge cases, etc

gnosis-info commented:
Travis automatic deployment:
 https://pr1164--dexreact.review.gnosisdev.com

W3stside commented:
@alfetopito careful on the height of the div considering the added content aka price suggestion. If it gets too tall (e.g when the user puts in too much sell and gets the over balance warning) the submit button will wrap to the right and look shitty af

W3stside commented on src/components/TradeWidget/PriceEstimations.tsx:
```diff
@@ -0,0 +1,89 @@
+
+  return (
+    <div
+      style={{ display: 'flex', justifyContent: 'space-between', padding: '0.1em 0 0.1em 0.5em', fontSize: '1.4rem' }}
```
You know what to do here

alfetopito replied:
Yup :)

Silver-tape-quick-hacky-fix-first-revision

W3stside commented on src/components/TradeWidget/index.tsx:
```diff
@@ -506,6 +499,12 @@ const TradeWidget: React.FC = () => {
 
+  const [priceShown, setPriceShown] = useState<'INVERSE' | 'DIRECT'>('INVERSE')
+
+  const swapPrices = (): void => {
```
Can be one liner lambda

alfetopito commented:
> @alfetopito careful on the height of the div considering the added content aka price suggestion. If it gets too tall (e.g when the user puts in too much sell and gets the over balance warning) the submit button will wrap to the right and look shitty af

Indeed. In that case, what's the best way to handle it? Just increase the default max height?

alfetopito commented:
> Other than a minor things, like the INFINITE, and probably the tooltip, I think it's good to go.
> 
> The others we can add it to our backlog. Feel free to continue working on them since you are on fire
> But I think we should freeze one version with what we have and deliver

@anxolin addressed your comments. I'll wait for your feedback on the tooltip text before merging, though.
